tests/xrdp/test_xrdp_region.c

```diff
@@ -0,0 +1,171 @@
+/**
+ * xrdp: A Remote Desktop Protocol server.
+ *
+ * Copyright (C) Jay Sorg, Christopher Pitstick 2004-2023
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Test driver for XRDP routines
+ *
+ * If you want to run this driver under valgrind to check for memory leaks,
+ * use the following command line:-
+ *
+ * CK_FORK=no valgrind --leak-check=full --show-leak-kinds=all \
+ *     .libs/test_xrdp
+ *
+ * without the 'CK_FORK=no', memory still allocated by the test driver will
+ * be logged
+ */
+
+#if defined(HAVE_CONFIG_H)
+#include "config_ac.h"
+#endif
+
+#include "log.h"
+#include "os_calls.h"
+#include <stdlib.h>
+
+#if defined(XRDP_PIXMAN)
+#include <pixman.h>
+#else
+#include "pixman-region.h"
+#endif
+
+#include "test_xrdp.h"
+#include "xrdp.h"
+#include <stdarg.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <setjmp.h>
+#ifdef __cplusplus
+// cmocka.h lacks c++ guards for CI
+extern "C" {
+#include <cmocka.h>
+}
+#define C_LINKAGE extern "C"
+#else
```

Comment:
I'm pretty sure we should be removing all the C+ stuff. How about having an `#error "cmocka is not supported with C++"` if `__cplusplus` is defined? Then all the C_LINKAGE stuff can go too.

Reply:
Updated.